### PR TITLE
 Allow configuring the "extra.symfony.root-dir" of the Symfony app 

### PR DESCRIPTION
**Pull request by nicolas-grekas**
Fixes https://github.com/symfony/flex/issues/243
and makes path management a bit more flexible.

### CODE REVIEW

File: src/Configurator/CopyFromPackageConfigurator.php [hunk 1]
```diff
@@ -97,7 +97,7 @@ public function copyFile(string $source, string $target, bool $overwrite = false
             throw new LogicException(sprintf('File "%s" does not exist!', $source));
         }
 
-        copy($source, $target);
+        file_put_contents($target, $this->options->expandTargetDir(file_get_contents($source)));
```
Comment (nicolas-grekas) [inline src/Configurator/CopyFromPackageConfigurator.php]: This replaces eg `%CONFIG_DIR%` in files copied from packages (same for recipes below).
Will allow replacing `config/bootstrap.php` by `%CONFIG_DIR%/bootstrap.php` in a few months when everyone will have moved to a more recent flex version.

File: src/Flex.php [hunk 2]
```diff
@@ -629,14 +629,17 @@ private function fetchRecipes(): array
...
+
         $options = array_merge([
             'bin-dir' => 'bin',
             'conf-dir' => 'conf',
             'config-dir' => 'config',
             'src-dir' => 'src',
             'var-dir' => 'var',
             'public-dir' => 'public',
-        ], $this->composer->getPackage()->getExtra());
+            'root-dir' => $extra['symfony']['root-dir'] ?? '.',
```
Comment (Pierstoval) [inline src/Flex.php]: How is `.` resolved when using it? Is it related to the `composer.json` file by composer or something?

Comment (nicolas-grekas) [reply src/Flex.php]: It's the same as using getenv()

Comment (Pierstoval) [reply src/Flex.php]: You mean `getcwd()` I guess 😄

**Pull request merged**